Civi/ActionSchedule/RecipientBuilder.php

```diff
@@ -314,7 +314,7 @@ protected function buildAddlRepeatPass() {
 
     $addlCheck = \CRM_Utils_SQL_Select::from($query['casAddlCheckFrom'])
       ->select('*')
-      ->merge($query, array('wheres'))// why only where? why not the joins?
+      ->merge($query, array('params', 'wheres'))// why only where? why not the joins?
```

Comment:
This is for point 1 - fixed by passing `params` so that `merge()` assigns it from the query object.